propagation_ot.go

```diff
@@ -137,50 +139,26 @@ func (p *binaryPropagator) Inject(
 	if !ok {
 		return opentracing.ErrInvalidCarrier
 	}
-	var err error
-	var sampledByte byte
-	if sc.raw.Sampled {
-		sampledByte = 1
-	}
 
-	// Handle the trace and span ids, and sampled status.
-	err = binary.Write(carrier, binary.BigEndian, sc.raw.TraceID)
-	if err != nil {
-		return err
-	}
-
-	err = binary.Write(carrier, binary.BigEndian, sc.raw.SpanID)
-	if err != nil {
-		return err
-	}
+	state := wire.TracerState{}
+	state.TraceId = sc.raw.TraceID
+	state.SpanId = sc.raw.SpanID
+	state.Sampled = sc.raw.Sampled
+	state.BaggageItems = sc.raw.Baggage
 
-	err = binary.Write(carrier, binary.BigEndian, sampledByte)
+	b, err := proto.Marshal(&state)
 	if err != nil {
 		return err
 	}
 
-	// Handle the baggage.
-	err = binary.Write(carrier, binary.BigEndian, int32(len(sc.raw.Baggage)))
-	if err != nil {
+	// Write the length of the marshalled binary to the writer.
+	len := uint32(len(b))
+	if err := binary.Write(carrier, binary.BigEndian, &len); err != nil {
```

Comment:
Use `binary.PutVarint` instead of `Write` and you can avoid throwing `&len` on the stack (where it likely lives now).